sdk-setup/src/mb2

```diff
@@ -1389,6 +1450,81 @@ run_prepare() {
     return $rc
 }
 
+cmake_post_cleanup() {
+    # Handle path changes for Qt Creator
+    local cmakefiles="CMakeCache.txt"
+
+    cp -p CMakeCache.txt CMakeCache.txt.qtccache
+
+    for cbpfile in *.cbp; do
+        [ -f "$cbpfile" ] || continue
+        cp -p "$cbpfile" "$cbpfile.qtccache"
+        cmakefiles="$cmakefiles $cbpfile"
+    done
+
+    if [[ "$STATEDIR" == "/home/mersdk/share"* && $MER_SSH_HOST_SHARED_HOME ]]; then
+        sed -i "s|/home/mersdk/share|$MER_SSH_HOST_SHARED_HOME|g" $cmakefiles
+    elif [[ "$STATEDIR" == "/home/src1"* && $MER_SSH_HOST_SHARED_SRC ]]; then
+        sed -i "s|/home/src1|$MER_SSH_HOST_SHARED_SRC|g" $cmakefiles
+    else
+        echo "Failed to set CMake paths for Qt Creator"
+        rm *.qtccache
+    fi
+}
+
+cmake_pre_setup() {
+    # Undo path changes for Qt Creator
+    if [[ -f "CMakeCache.txt.qtccache" ]]; then
+        mv CMakeCache.txt.qtccache CMakeCache.txt
+        for cbp_file in *.cbp.qtccache; do
+            mv $cbp_file ${cbp_file%.qtccache}
+        done
+    fi
+}
+
+run_cmake() {
+    local rpmbuild="--build . --target all"
+    local sb2args=( -t "$OPT_TARGET" env PATH="${WRAPPERSDIR}:${USERWRAPPERSDIR}:${PATH}" )
+
+    # CMake projects need to have dependencies installed before running
+    # any CMake commands
+    if [[ $OPT_SPEC ]]; then
+        verify_target_dependencies
+    fi
+
+    if [[ " $* " == *" $rpmbuild "* && $OPT_SPEC ]]; then
+        local build_tgt=$(sb2 -t $OPT_TARGET gcc -dumpmachine)
+        build_tgt=${build_tgt%-gnueabi}
+
+        cp -a "$OPT_SPEC" "$TMP_SPEC"
+
+        # get rid of potential windows linefeeds
+        sed -i "s/\x0D$//g" "$TMP_SPEC"
+
+        fix_package_version "$TMP_SPEC"
+
+        eval "set -- $(printf "%q " "$@" |sed "s/ $rpmbuild / /")"
+        init_cmake_wrapper $WRAPPERSDIR/cmake "$@" "$OPT_SRC_DIR"
+        init_simple_wrapper $WRAPPERSDIR/make
+
+        (
+            cd_to_spec_setup_dir
+            
+            echo sb2 "${sb2args[@]}" rpmbuild --build-in-place --target=$build_tgt --define "noecho 1" \
+                --define "_sourcedir $(dirname $OPT_SPEC)" -bc --short-circuit "$TMP_SPEC"
+            
+            sb2 "${sb2args[@]}" rpmbuild --build-in-place --target=$build_tgt --define "noecho 1" \
+                --define "_sourcedir $(dirname $OPT_SPEC)" -bc --short-circuit "$TMP_SPEC"
+    
+        )
+    else
+        init_cmake_wrapper $WRAPPERSDIR/cmake "$@"
+        sb2 "${sb2args[@]}" cmake
+    fi
+    local rc=$?
+    return $rc
```

Comment:
I would drop all occurrences of this. It is likely there was some need to this before but now all occurrences seems to be without effect.

Reply:
To me it looks like the idea has been to have mb2 exit with the exit status of the executed command. Now this is broken because we do cmake_post_cleanup after the command. Perhaps I should just fix it instead of dropping?

Reply:
The last *executed* command's exit code defines the exit code of the function, it is not necessary to capture and return it explicitly. I guess there were some additional actions between `local rc=$?` and `return $rc` before, which are gone now, so the extra step is not needed anymore.

You are right cmake_post_cleanup after the command needs to be fixed.

> Perhaps I should just fix it instead of dropping?

So I would recommend both. Current occurrences dropped and one added near the cmake_post_cleanup call.